applications/serial_lte_modem/src/mqtt_c/slm_at_mqtt.c

```diff
@@ -370,7 +418,7 @@ static int do_mqtt_disconnect(void)
 	int err;
 
 	if (!ctx.connected) {
-		return -ENOTCONN;
+		return -EAGAIN;
```

Comment:
EAGAIN indicates that the operation would succeed at later point of time, which is not the case in here, if we have to close the connection and reopen it. Same for others.

Reply:
In libc errno EAGAIN means no more context, i.e. wrong context in my understanding.
Anyway any errno will do as we only want to return ERROR for the AT command.

Reply:
Generally, EAGAIN and it's sister code EWOULDBLOCK are used when something would fail, but could succeed in other conditions. I think that in this case, if the user needs to connect again, it would be better to use previous -ENOTCONN or other error that indicates that we are not connected/have disconnected.

Reply:
@junqingzou: Please change this to ENOTCONN or other value that indicates not connected / disconnected.